### PR TITLE
Adding cli to custom klipper helm image

### PR DESCRIPTION
**Pull request by vitorsavian**
#### Proposed Changes ####

A new cli flag for changing our klipper helm image that is default

#### Types of Changes ####

New feature

#### Verification ####

```shell
drone exec --trusted --pipeline amd64
``` 

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/3865

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: pkg/cli/cmds/server.go [hunk 1]
```diff
@@ -252,6 +253,11 @@ var ServerFlags = []cli.Flag{
 		Destination: &ServerConfig.KubeConfigMode,
 		EnvVar:      version.ProgramUpper + "_KUBECONFIG_MODE",
 	},
+	&cli.StringFlag{
+		Name:        "kippler-helm-image",
+		Usage:       "(helm) Image for our helm controller",
+		Destination: &ServerConfig.KlipperHelmImage,
```
Comment (brandond) [inline pkg/cli/cmds/server.go:259]:
```suggestion
		Name:        "helm-job-image",
		Usage:       "(helm) Default image to use for helm controller jobs",
		Destination: &ServerConfig.HelmJobImage,
```

File: pkg/cli/server/server.go [hunk 2]
```diff
@@ -113,6 +113,7 @@ func run(app *cli.Context, cfg *cmds.Server, leaderControllers server.CustomCont
 	serverConfig.ControlConfig.DataDir = cfg.DataDir
 	serverConfig.ControlConfig.KubeConfigOutput = cfg.KubeConfigOutput
 	serverConfig.ControlConfig.KubeConfigMode = cfg.KubeConfigMode
+	serverConfig.ControlConfig.KlipperHelmImage = cfg.KlipperHelmImage
```
Comment (brandond) [inline pkg/cli/server/server.go:116]:
```suggestion
	serverConfig.ControlConfig.HelmJobImage = cfg.HelmJobImage
```

File: pkg/daemons/config/types.go [hunk 3]
```diff
@@ -166,6 +166,7 @@ type Control struct {
 	ServiceNodePortRange     *utilnet.PortRange
 	KubeConfigOutput         string
 	KubeConfigMode           string
+	KlipperHelmImage         string
```
Comment (brandond) [inline pkg/daemons/config/types.go:169]:
```suggestion
	HelmJobImage         string
```

File: pkg/server/server.go [hunk 4]
```diff
@@ -211,6 +211,14 @@ func coreControllers(ctx context.Context, sc *Context, config *Config) error {
 		helmchart.DefaultJobImage = config.ControlConfig.SystemDefaultRegistry + "/" + helmchart.DefaultJobImage
 	}
 
+	if config.ControlConfig.KlipperHelmImage != "" {
+		helmchart.DefaultJobImage = config.ControlConfig.KlipperHelmImage
+	}
+
+	if config.ControlConfig.SystemDefaultRegistry != "" && config.ControlConfig.KlipperHelmImage != "" {
+		helmchart.DefaultJobImage = config.ControlConfig.SystemDefaultRegistry + "/" + config.ControlConfig.KlipperHelmImage
+	}
```
Comment (brandond) [inline pkg/server/server.go:220]: Only apply the system default registry if a custom job image hasn't been set; the custom job image should take precedence.
```suggestion
	if config.ControlConfig.HelmJobImage != "" {
		helmchart.DefaultJobImage = config.ControlConfig.HelmJobImage
	} else if config.ControlConfig.SystemDefaultRegistry != "" {
		helmchart.DefaultJobImage = config.ControlConfig.SystemDefaultRegistry + "/" + helmchart.DefaultJobImage
	}
```

**Review by brandond — CHANGES_REQUESTED** [review]
Although the images are still called klipper-helm and klipper-lb, we don't call things "klipper" any more, even internally. Klipper was an early project name for K3s, and we've completely dropped except for these two image names.

File: pkg/cli/cmds/server.go [hunk 5]
```diff
@@ -252,6 +253,11 @@ var ServerFlags = []cli.Flag{
 		Destination: &ServerConfig.KubeConfigMode,
 		EnvVar:      version.ProgramUpper + "_KUBECONFIG_MODE",
 	},
+	&cli.StringFlag{
+		Name:        "helm-job-image",
+		Usage:       "(helm) Default image to use for helm controller",
```
Comment (brandond) [inline pkg/cli/cmds/server.go:258]: clarify that this is for helm jobs, not the controller. the controller that creates jobs runs in the main k3s process.
```suggestion
		Usage:       "(helm) Default image to use for helm jobs",
```

**Review by brandond — CHANGES_REQUESTED** [review]
nit, lgtm otherwise!

**Review by brandond — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]

**Pull request merged**